integrations/beacon-blobs/src/runtime.ts

```diff
@@ -0,0 +1,175 @@
+      config.executionRPC = process.env.KYVEJS_BEACON_BLOBS_EXECUTION_RPC;
+
+      console.log("set config executionRPC to", config.executionRPC)
+    }
+
+    if (process.env.KYVEJS_BEACON_BLOBS_CONSENSUS_RPC) {
+      config.consensusRPC = process.env.KYVEJS_BEACON_BLOBS_CONSENSUS_RPC;
+
+      console.log("set config consensusRPC to", config.consensusRPC)
+    }
+
+
+    this.config = config;
+  }
+
+  async  getDataItem(_: Validator, key: string): Promise<any> {
+    const provider = new providers.StaticJsonRpcProvider(this.config.executionRPC);
+
+    const currentHeight = await provider.getBlockNumber();
+
+    const hexKey = hexValue(+key);
+
+    const block = await provider.getBlockWithTransactions(hexKey);
+
+    // only validate if current height is already 'finalized'
+    if (block.number >= currentHeight - 256) {
+      throw new Error(
+        `Finality not reached yet; waiting for next block`
+      )
+    }
+
+    // Get all type3 transactions that has been sent to the sequencer inbox
+    const filteredTransactions = block.transactions.filter(
+      (tx) => {
+        if (tx.to) {
+          return tx.type == 3 && this.config.sequencer.includes(tx.to)
+        }
+        return false
+      }
+    );
+
+    let type3TxsToSequencer: string[] = [];
+    for (const tx of filteredTransactions) {
+      const txDetail = await getTransactionByHash(this.config.executionRPC, tx.hash);
+      txDetail["blobVersionedHashes"].forEach((bHash: any) => type3TxsToSequencer.push(bHash))
+    }
+
+    let blobs: any;
+
+    // Calculate corresponding slot number
+    const slotNumber = (block.timestamp - this.config.genesisTime) / 12
+
+    await axios.get(`${this.config.consensusRPC}/eth/v1/beacon/blob_sidecars/${slotNumber}`, {
+      headers: {
+        'accept': 'application/json'
+      }
+    }).then((res) => {
+      blobs = res.data.data;
+    }).catch(err => {
+      throw new Error(
+        `Failed to query '/eth/v1/beacon/blob_sidecars/${slotNumber}'`
+      );
+    });
+
+    let includedBlobs: any[] = [];
+    // For each blob, take KZG commitment and check if it matches with a first versioned hash of type3TxToSequencer.
+    // This means that the blob was actually sent to the sequencer address.
+    // If it matches, include the blob in data item, if not, skip blob.
+    blobs.forEach((b: any) => {
+      const commitment= createVersionedHash(b["kzg_commitment"]);
+
+      if (type3TxsToSequencer.includes(commitment)) {
+        includedBlobs.push(b)
+      }
+    });
+
+    if (includedBlobs.length != type3TxsToSequencer.length) {
+      throw new Error(
+        `Length of included blobs and txs to sequencer is not equal`
+      )
+    }
+
+    return {
+      key,
+      value: {
+        slot: slotNumber,
+        blobs: includedBlobs
+      },
+    };
+  }
+
+
+  async prevalidateDataItem(_: Validator, item: DataItem): Promise<boolean> {
+    // check if item value is not null
```

Comment:
a null check does not take place, this would be a null check: `return item.value && item.value.slot && item.value.includedBlobs`